src/new/templates/rust/chat/chat/src/lib.rs

```diff
@@ -22,51 +31,238 @@ enum ChatResponse {
+                channel_id.clone(),
+                WsMessageType::Text,
+                payload,
+            )?;
+        }
+        ChatRequest::History => {
+            // If this is an HTTP request, send a response to the http server
+
+            Response::new()
+                .ipc(
+                    serde_json::to_vec(&ChatResponse::History {
+                        messages: message_archive.clone(),
+                    })
+                    .unwrap(),
+                )
+                .send()
+                .unwrap();
+        }
+    };
+
+    Ok(())
+}
+
+fn handle_message(
     our: &Address,
     message_archive: &mut MessageArchive,
+    channel_id: &mut u32,
 ) -> anyhow::Result<()> {
     let message = await_message().unwrap();
 
     match message {
         Message::Response { .. } => {
-            print_to_terminal(0, &format!("{package_name}: unexpected Response: {:?}", message));
-            panic!("");
-        },
-        Message::Request { ref source, ref ipc, .. } => {
-            match serde_json::from_slice(ipc)? {
-                ChatRequest::Send { ref target, ref message } => {
-                    if target == &our.node {
-                        print_to_terminal(0, &format!("{package_name}|{}: {}", source.node, message));
-                        message_archive.push((source.node.clone(), message.clone()));
-                    } else {
-                        let _ = Request::new()
-                            .target(Address {
-                                node: target.clone(),
-                                process: ProcessId::from_str("{package_name}:{package_name}:{publisher}")?,
-                            })
-                            .ipc(ipc.clone())
-                            .send_and_await_response(5)?
-                            .unwrap();
-                    }
-                    Response::new()
-                        .ipc(serde_json::to_vec(&ChatResponse::Ack).unwrap())
-                        .send()
-                        .unwrap();
-                },
-                ChatRequest::History => {
-                    Response::new()
-                        .ipc(serde_json::to_vec(&ChatResponse::History {
-                            messages: message_archive.clone(),
-                        }).unwrap())
-                        .send()
-                        .unwrap();
-                },
-            }
-        },
+            print_to_terminal(0, &format!("{package_name}: got response - {:?}", message));
+            return Ok(());
+        }
+        Message::Request {
+            ref source,
+            ref ipc,
+            ..
+        } => {
+            // Requests that come from other nodes running this app
+            handle_chat_request(our, message_archive, channel_id, source, &ipc, false)?;
+            // Requests that come from our http server
+            handle_http_server_request(our, message_archive, source, ipc, channel_id)?;
```

Comment:
Small change: can we have `handle_chat_request()` & `handle_http_server_request()` have same order of arguments? I.e. move `handle_http_server_request()`s `channel_id` argument to be the third argument, same as `handle_chat_request()` ?